protocol.md

```diff
@@ -568,6 +568,127 @@ Tus-Resumable: 1.0.0
 Upload-Offset: 11
 ```
 
+### Partial Checksum
+
+This extension allows Clients that use the Checksum extension to verify and confirm the
+integrity of chunks that were not received in full by the Server - thereby avoiding that
+intact data needs to be sent again.
+
+Servers that support this extension MUST add `partial-checksum` to the `Tus-Extension` header.
+
+If a Server receives an incomplete `PATCH` or `POST` request with `Upload-Checksum` header, it
+MAY store the incompletely received chunk's data (hereafter referred to as *partial data*) in
```

Comment:
You're absolutely right. I'll replace it with a SHOULD in the next commit. I'd like to avoid MUST to allow situational flexibility on the server side, i.e. if the server can encounter situations where it can't store the partial data - or wants to impose a quota or size limit on partial data.